crates/uv-resolver/src/lock/mod.rs

```diff
@@ -3245,11 +3246,16 @@ impl PackageWire {
                 if *version != wheel.filename.version
                     && *version != wheel.filename.version.clone().without_local()
                 {
-                    return Err(LockError::from(LockErrorKind::InconsistentVersions {
-                        name: self.id.name,
-                        version: version.clone(),
-                        wheel: wheel.clone(),
-                    }));
+                    if !matches!(
+                        parse_boolish_environment_variable(EnvVars::UV_SKIP_WHEEL_FILENAME_CHECK),
+                        Ok(Some(true))
```

Comment:
I would prefer to do this once and pass it around, but it's legitimately annoying to do here, because this is ultimately triggered by a `TryFrom` on `LockWire`, so even if we did have it as a setting, we'd have to restructure the Serde setup to make it work.

Honestly, I kind of think we should parse these once (`EnvironmentOptions`) then set some global flags on a singleton that we can then access anywhere.

Reply:
I tried this out here: https://github.com/astral-sh/uv/pull/16047